### ISSUE TITLE
Invalid CycloneDX SHA1 algorithm

### ISSUE THREAD

**Issue by julien-carsique-sonarsource** [anchore/syft#1001]
**What happened**:

The SBOM may contain an invalid algorithm name in `.components[].externalReferences[].hashes[].alg`.
The wrong value does not match [the reference](https://cyclonedx.org/docs/1.4/json/#components_items_externalReferences_items_hashes_items_alg).

```bash
$ syft packages IMAGE -o cyclonedx-json=/tmp/bom.json
$ cyclonedx validate --input-file /tmp/bom.json
Unable to validate against any JSON schemas.
BOM is not valid.

# cyclone merge and convert commands give a pointer to the error
$ cyclone convert --input-file /tmp/bom.json ...
Unhandled exception: System.Text.Json.JsonException: The JSON value could not be converted to CycloneDX.Models.Hash+HashAlgorithm. Path: $.components[153].externalReferences[0].hashes[0].alg ...

$ jq ".components[153].externalReferences[].hashes[].alg" /tmp/bom.json
"sha1"
```

**What you expected to happen**:

```shell
$ cyclonedx validate --input-file /tmp/bom.json
BOM validated successfully.
```

**How to reproduce it (as minimally and precisely as possible)**:

Use `cyclonedx validate` on a CycloneDX SBOM containing a sha1 alg.

Replacing the `sha1` value with `SHA-1` is fixing the issue.

**Anything else we need to know?**:

**Environment**:
- Output of `syft version`:
```
Application:        syft
Version:            0.46.0
JsonSchemaVersion:  3.2.3
BuildDate:          2022-05-12T16:12:28Z
GitCommit:          91e2fd8532254216a83c80891611751d0c8fba7b
GitDescription:     v0.46.0
Platform:           linux/amd64
GoVersion:          go1.18.1
Compiler:           gc
```
- OS (e.g: `cat /etc/os-release` or similar):
Ubuntu 20.04.4 LTS

**Comment by spiffcs** [comment]
Thanks @julien-carsique-sonarsource for the issue! We added new checks to make sure this does not happen again and will be releasing a patch today:
https://github.com/anchore/syft/pull/1000

**Comment by spiffcs** [comment]
@julien-carsique-sonarsource we have released `v0.46.1`. Let me know if you're still running into the issue surrounding the invalid `sha1` value. We've also added integration checks against the `cyclonedx validate` command so we can catch any more of these in the future.

**Comment by spiffcs** [comment]
Closing this after testing and validating the newest release. If there is still an issue after this I can reopen this thread so we can discuss.
